rfcs/text/remove-support-for-uncontrolled-inputs.md

```diff
@@ -0,0 +1,233 @@
+
+We also already provide various codemods in our [Carbon Codemod](https://github.com/Sage/carbon-codemod) repo that could be used by our consumers to assist with the move to a controlled component. Examples of this are:
+- A [rename-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/rename-prop) codemod that consumers could use to change a `defaultValue` value prop to `value`.
+- A [add-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/add-prop) codemod that could be used to add to a `value` prop.
+- A [remove-prop](https://github.com/Sage/carbon-codemod/tree/master/transforms/remove-prop) codemod that could be used to remove a `defaultValue` prop.
+- A [replace-prop-value](https://github.com/Sage/carbon-codemod/tree/master/transforms/replace-prop-value) codemod that changes the value of a prop.
+
+There is also support for writing controlled components in React's documentation. This can be found [here](https://reactjs.org/docs/forms.html#controlled-components).
+
+# How we teach this
+
+This RFC will be communicated in the Sage `#carbon` and `#carbon-announcement` slack channels, where we will ask our consumers to feedback on the RFC. Beyond that, it will be visible in the release notes when consumers upgrade. This RFC document will reside in the repository for future reference.
+
+We will be able to support consumers of Carbon in the `#carbon` slack channel too. 
+
+# Documentation
+
+As part of this work we will make sure that the documentation reflects what we will support. We will update the component description to include that we specifically only support controlled usage of these components. We will also ensure Storybook examples are updated where any uncontrolled behaviours are demonstrated. This will involve removing any uncontrolled examples and ensuring we have the controlled examples.
+
+# Further Considerations
+
+If/when we introduce a `FileInput` component into Carbon, this will have to be an uncontrolled component. This is because `<input type="file" />` can only be uncontrolled as [the value can only be specified by the user and not be done programmatically](https://reactjs.org/docs/uncontrolled-components.html#the-file-input-tag). As a file input only works uncontrolled, we would not need to maintain any controlled behaviour.
```

Comment:
The last sentence here is a little confusing, and I'm not sure it adds any value anyway - I think the previous 2 sentences and accompanying link say it all. If we do keep it I prefer a phrasing such as `As file inputs cannot be controlled, the Carbon FileInput component would have to be uncontrolled.`